File: code_examples/5-Holding_Data/4-Associative_Array/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a nested if statement

#include <stdio.h>

int main() {
    // this function uses a nested if statement
    int cookiesSold;

    // input
    printf("Enter the number of boxes of cookies you sold: ");
    scanf("%d", &cookiesSold);
    printf("\n");

    // process & output
    if (cookiesSold >= 20) {
        if (cookiesSold < 30) {
            printf("You get a small prize.\n");
        } else {
            printf("You get a large prize.\n");
        }
    } else {
        printf("No prize.\n");
    }

    printf("\nDone.\n");
    return 0;
}

File: code_examples/5-Holding_Data/4-Associative_Array/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module uses an associative array
"""


def main() -> None:
    """The main() this function function uses an associative array, returns None."""

    airports = {}
    some_info = {'item1': 1,
                'item2': 2,
                }

    # adding items
    airports['YYZ'] = "Toronto Pearson"
    airports['YOW'] = "Ottawa Canada"
    airports['DUB'] = "Dublin Ireland"
    airports['LHR'] = "London Heathrow"

    # input & process

    print("All the airports:")
    for key, value in airports.items():
        print(f"The airport code is {key} for {value}.")
    print("")

    airport_name = input("Type in an airport code: ")
    if airport_name in airports:
        print(f"The name of the airport you chose is {airports[airport_name]}.")
    else:
        print("That airport is not in the airport's dictionary.")

    print("\nDone.")


if __name__ == "__main__":
    main()
